plugins/ci/pkg/ci/trivy.go

```diff
@@ -138,19 +140,35 @@ func downloadMissingImages(folderPath string, dockerImages []trivymodels.DockerI
 	return nil
 }
 
-func downloadImageViaSkopeo(folderPath, imageName string) error {
+func downloadImageViaSkopeo(cmdExecutor cmdExecutor, folderPath, imageName string, rc *models.RegistryCredential) (string, error) {
 	logrus.Infof("Downloading missing image %s", imageName)
 	dockerURL := "docker://" + imageName
 	archiveName := "docker-archive:" + folderPath + clearString(imageName)
-	cmd := exec.Command("skopeo", "copy", dockerURL, archiveName)
+	args := []string{"copy"}
+
+	if rc != nil {
```

Comment:
In my tests - using registry credentials for public images worked, is there any concern about it?